components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthenticationService.java

```diff
@@ -266,6 +266,11 @@ private String getErrorCode(AuthServiceResponseWrapper response) throws AuthServ
     private String getErrorMessage(AuthServiceResponseWrapper response) throws AuthServiceException {
 
         Map<String, String> queryParams = AuthServiceUtils.extractQueryParams(response.getRedirectURL());
+
+        if (queryParams.containsKey(AuthServiceConstants.PASSWORD_EXPIRED_PARAM)
```

Comment:
Why did we decide to treat this as a special error param and message without utilizing the existing AUTH_FAILURE_MSG_PARAM ?

Reply:
Currently, both the **AUTH_FAILURE_MSG_PARAM** and **AUTH_FAILURE_PARAM** are not included as query parameters for password expiry. Instead, the **PASSWORD_EXPIRED_QUERY_PARAMS** is used [[1]](https://github.com/wso2-extensions/identity-governance/pull/855/files#diff-851cea81cf897e5fa5930f19a76db40c46dc4ca687405d03e3cfdc5d3171e360R177). 

We could add the existing Auth Failure parameters to the query, however, this could potentially disrupt existing implementations that rely on these parameters. Instead, we decided to utilize the already existing **PASSWORD_EXPIRED_QUERY_PARAMS** and add the **PASSWORD_EXPIRED_MSG_PARAM**.